serial-settings/src/lib.rs

```diff
@@ -122,163 +116,377 @@ impl<'a, P: Platform<Y>, const Y: usize> Interface<'a, P, Y> {
+                    Ok(Some(_stored)) => match interface
+                        .platform
+                        .clear(interface.buffer, key.as_bytes())
+                    {
+                        Ok(()) => {
+                            writeln!(
+                                interface,
+                                "Clear stored `{}`",
+                                key.as_str()
+                            )
+                        }
+                        Err(e) => {
+                            writeln!(
+                                interface,
+                                "Failed to clear `{}` from storage: {e:?}",
+                                key.as_str()
+                            )
+                        }
+                    }
+                    .unwrap(),
+                }
+            },
+        );
+        interface.updated = true;
+        writeln!(interface, "Some values may require reboot to become active")
+            .unwrap();
     }
 
-    fn handle_save(
+    fn handle_store(
         _menu: &menu::Menu<Self, P::Settings>,
         item: &menu::Item<Self, P::Settings>,
         args: &[&str],
         interface: &mut Self,
         settings: &mut P::Settings,
     ) {
-        match interface.platform.save(interface.buffer, menu::argument_finder(item, args, "item").unwrap(), settings) {
-            Ok(_) => writeln!(
-                interface,
-                "Settings saved. You may need to reboot for the settings to be applied"
-            )
-            .unwrap(),
-            Err(e) => {
-                writeln!(interface, "Failed to save settings: {e:?}").unwrap()
-            }
-        }
+        let key = menu::argument_finder(item, args, "path").unwrap();
+        Self::iter_root(
+            key,
+            interface,
+            settings,
+            |key, interface, settings, defaults| {
+                // Get default value checksum
+                let slic = postcard::ser_flavors::Slice::new(interface.buffer);
+                // Could in theory serialize directly into the hasher
+                let mut check = match defaults.get_postcard_by_key(key, slic) {
+                    Ok(slic) => yafnv::fnv1a::<u32>(slic),
+                    Err(miniconf::Error::Traversal(Traversal::Absent(
+                        _depth,
+                    ))) => {
+                        log::warn!("Default absent: `{}`", key.as_str());
+                        return;
+                    }
+                    Err(e) => {
+                        writeln!(
+                            interface,
+                            "Failed to get `{}` default: {e:?}",
+                            key.as_str()
+                        )
+                        .unwrap();
+                        return;
+                    }
+                };
+
+                // Get stored value checksum
+                match interface.platform.fetch(interface.buffer, key.as_bytes())
```

Comment:
Why are we mixing postcard above in `defaults`, but then postcard doesn't show up at all in the `fetch` API? Does this assume that its always postcard-serialized?

Reply:
now serial-settings decides about key and value format.
it only needs to be internally consistent. it doesn't leak that info.